Amplify/DevMenu/AmplifyDevMenu.swift

```diff
@@ -9,39 +9,38 @@ import Foundation
 import UIKit
 import SwiftUI
 
+/// Presents a developer menu using the provided `DevMenuPresentationContextProvider`
+/// upon notification from a `TriggerRecognizer`. Default recognizer is a `LongPressGestureRecognizer`
 @available(iOS 13.0.0, *)
-public final class AmplifyDevMenu: TriggerDelegate {
+public final class AmplifyDevMenu: DevMenuBehavior, TriggerDelegate {
 
-    weak var devMenuDelegate: DevMenuDelegate?
-    var longPressGestureRecognizer: LongPressGestureRecognizer?
+    weak var devMenuPresentationContextProvider: DevMenuPresentationContextProvider?
+    var triggerRecognizer: TriggerRecognizer?
 
-    init(delegate: DevMenuDelegate) {
-        self.devMenuDelegate = delegate
-        self.longPressGestureRecognizer = LongPressGestureRecognizer(
-                                                uiWindow: delegate.presentationContext(),
-                                                triggerDelegate: self)
+    init(devMenuPresentationContextProvider: DevMenuPresentationContextProvider) {
+        self.devMenuPresentationContextProvider = devMenuPresentationContextProvider
+        self.triggerRecognizer = LongPressGestureRecognizer(
+            uiWindow: devMenuPresentationContextProvider.devMenuPresentationContext(),
+            triggerDelegate: self)
```

Comment:
Remove the triggerDelegate parameter from init. We can make use of the delegate method to update it. So the above code would look like this:
```
self.triggerRecognizer = LongPressGestureRecognizer(
            uiWindow: devMenuPresentationContextProvider.devMenuPresentationContext())
self.triggerRecognizer.updateTriggerDelegate(self)
```